File: android/EpubRnD/src/com/hurix/epubRnD/VOs/PageVO.java
package com.hurix.epubRnD.VOs;

import com.hurix.epubRnD.Constants.GlobalConstants;

public class PageVO 
{
	private ChapterVO _chapterVO;
	private int indexOfPage;
	private int indexOfChapter;
	private int firstWordID;
	private int lastWordID;
	private int wordIDToGetIndexOfPage;
	
	public ChapterVO getChapterVO() {
		return _chapterVO;
	}

	public void setChapterVO(ChapterVO _chapterVO) {
		this._chapterVO = _chapterVO;
	}

	public int getIndexOfPage() {
		return indexOfPage;
	}
	/**
	 * when we don't know the index of page then set it to GlobalConstants.GET_PAGE_INDEX_USING_WORD_ID
	 * and set any wordID belong to respective page
	 * @param indexOfPage
	 */
	public void setIndexOfPage(int indexOfPage) {
		this.indexOfPage = indexOfPage;
	}

	public int getIndexOfChapter() {
		return indexOfChapter;
	}

	public void setIndexOfChapter(int indexOfChapter) {
		this.indexOfChapter = indexOfChapter;
	}

	public int getFirstWordID() {
		return firstWordID;
	}

	public void setFirstWordID(int firstWordID) {
		this.firstWordID = firstWordID;
	}

	public int getLastWordID() {
		return lastWordID;
	}

	public void setLastWordID(int lastWordID) {
		this.lastWordID = lastWordID;
	}

	public int getWordIDToGetIndexOfPage() {
		return wordIDToGetIndexOfPage;
	}

	public void setWordIDToGetIndexOfPage(int wordIDToGetIndexOfPage) {
		this.wordIDToGetIndexOfPage = wordIDToGetIndexOfPage;
	}

}

File: ios/EpubRnD/EpubRnD/PageVO.m
//
//  PageVO.m
//  EpubRnD
//
//  Created by UdaySravan K on 10/06/14.
//  Copyright (c) 2014 hurix. All rights reserved.
//

#import "PageVO.h"

@implementation PageVO

- (void) setIndexOfPage:(NSInteger) indexOfPage
{
    mIndexOfPage = indexOfPage;
}

- (NSInteger) getIndexOfPage
{
    return mIndexOfPage;
}

- (void) setIndexOfChapter:(NSInteger) indexOfChapter
{
    mIndexOfChapter = indexOfChapter;
}

- (NSInteger) getIndexOfChapter
{
    return mIndexOfChapter;
}

- (void) setFirstWordID:(NSInteger) firstWordIDofPage
{
    firstWordID = firstWordIDofPage;
}
- (NSInteger) getFirstWordID
{
    return firstWordID ;
}
- (void) setLastWordID:(NSInteger) lastWordIDofPage
{
    lastWordID = lastWordIDofPage;
}
- (NSInteger) getLastWordID
{
    return lastWordID;
}

//- (void) setBookmarked:(BOOL) status
//{
//    isBookmarked = status;
//}
//- (BOOL) isBookmarked
//{
//    return isBookmarked;
//}

- (void)dealloc
{
    NSLog(@"my page view released");
}

- (id)initWithCoder:(NSCoder *)aDecoder
{
    self = [super init];
    if(self)
    {
        mIndexOfPage = ((NSNumber *)[aDecoder decodeObjectForKey:@"IndexOfPage"]).integerValue;
        mIndexOfChapter = ((NSNumber *)[aDecoder decodeObjectForKey:@"IndexOfChapter"]).integerValue;
        firstWordID = ((NSNumber *)[aDecoder decodeObjectForKey:@"FirstWordID"]).integerValue;
        lastWordID = ((NSNumber *)[aDecoder decodeObjectForKey:@"LastWordID"]).integerValue;
    }
    return self;
}

- (void)encodeWithCoder:(NSCoder *)aCoder
{
    [aCoder encodeObject:[NSNumber numberWithInteger:mIndexOfPage]  forKey:@"IndexOfPage"];
    [aCoder encodeObject:[NSNumber numberWithInteger:mIndexOfChapter]  forKey:@"IndexOfChapter"];
    [aCoder encodeObject:[NSNumber numberWithInteger:firstWordID]  forKey:@"FirstWordID"];
    [aCoder encodeObject:[NSNumber numberWithInteger:lastWordID]  forKey:@"LastWordID"];
}
@end
